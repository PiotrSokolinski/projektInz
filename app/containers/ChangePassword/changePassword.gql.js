import gql from 'graphql-tag'

export default gql`
  mutation changePassword($password: String!, $newPassword: String!) {
    changePassword(password: $password, newPassword: $newPassword) {
      id
    }
  }
`
