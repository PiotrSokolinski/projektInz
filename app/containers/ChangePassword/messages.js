/*
 * ChangePassword Messages
 *
 * This contains all the text for the ChangePassword container.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.containers.ChangePassword'

export default defineMessages({
  retypePasswordLabel: {
    id: `${scope}.retypePasswordLabel`,
    defaultMessage: 'Retype your new password ',
  },
  passwordEmptyError: {
    id: `${scope}.passwordEmptyError`,
    defaultMessage: 'Password is required',
  },
  passwordsDoNotMatchError: {
    id: `${scope}.passwordsDoNotMatchError`,
    defaultMessage: 'Passwords must match',
  },
  passwordConfirmEmpty: {
    id: `${scope}.passwordConfirmEmpty`,
    defaultMessage: 'Password confirm is required',
  },
  passwordLabel: {
    id: `${scope}.passwordLabel`,
    defaultMessage: 'Password',
  },
  repeatedPasswordLabel: {
    id: `${scope}.repeatedPasswordLabel`,
    defaultMessage: 'Retype password',
  },
  changePassword: {
    id: `${scope}.changePassword`,
    defaultMessage: 'Change password',
  },
})
