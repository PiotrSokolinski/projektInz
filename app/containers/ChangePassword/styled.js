import styled from 'styled-components'

import Button from 'components/Button'

export const ChangePasswordButton = styled(Button)`
  width: 250px;
`
