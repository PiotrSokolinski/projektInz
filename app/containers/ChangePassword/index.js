/**
 *
 * ChangePassword
 *
 */

import React from 'react'
import * as Yup from 'yup'
import PropTypes from 'prop-types'
import { FormattedMessage, injectIntl } from 'react-intl'
import { Formik } from 'formik'

import Input from 'components/Input'

import messages from './messages'
import { ChangePasswordButton } from './styled'

const initialValues = { password: '', repeatedPassword: '' }

const validationSchema = intl =>
  Yup.object().shape({
    password: Yup.string().required(intl.formatMessage(messages.passwordEmptyError)),
    repeatedPassword: Yup.string()
      .oneOf([Yup.ref('password'), null], intl.formatMessage(messages.passwordsDoNotMatchError))
      .required(intl.formatMessage(messages.passwordConfirmEmpty)),
  })

const ChangePassword = ({ intl }) => (
  <Formik onSubmit={() => {}} initialValues={initialValues} validationSchema={validationSchema(intl)}>
    {({ errors, touched, values, handleChange, handleBlur, handleSubmit }) => (
      <React.Fragment>
        <Input
          id="passwordField"
          label={intl.formatMessage(messages.passwordLabel)}
          error={touched.password && errors.password}
          inputProps={{
            type: 'password',
            name: 'password',
            value: values.password,
            onChange: handleChange,
            onBlur: handleBlur,
          }}
        />
        <Input
          id="repeatedPasswordField"
          label={intl.formatMessage(messages.repeatedPasswordLabel)}
          error={touched.repeatedPassword && errors.repeatedPassword}
          inputProps={{
            type: 'password',
            name: 'repeatedPassword',
            value: values.repeatedPassword,
            onChange: handleChange,
            onBlur: handleBlur,
          }}
        />
        <ChangePasswordButton type="submit" onClick={handleSubmit}>
          <FormattedMessage {...messages.changePassword} />
        </ChangePasswordButton>
      </React.Fragment>
    )}
  </Formik>
)

ChangePassword.propTypes = {
  intl: PropTypes.object,
}

export default injectIntl(ChangePassword)
