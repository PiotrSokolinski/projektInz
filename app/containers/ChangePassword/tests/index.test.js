/**
 *
 * Tests for ChangePassword
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import ChangePassword from '../index'
import { DEFAULT_LOCALE } from '../../../i18n'

describe('<ChangePassword />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error')
    render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <ChangePassword />
      </IntlProvider>,
    )
    expect(spy).not.toHaveBeenCalled()
  })

  it('Should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(
      <IntlProvider locale={DEFAULT_LOCALE}>
        <ChangePassword />
      </IntlProvider>,
    )
    expect(firstChild).toMatchSnapshot()
  })
})
