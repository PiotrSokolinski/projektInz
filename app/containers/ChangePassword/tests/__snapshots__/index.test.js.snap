// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChangePassword /> Should render and match the snapshot 1`] = `
<div
  class="field"
  id="passwordField"
>
  <label
    class="styled__StyledLabel-oaisvy-0 fCsvBu"
  >
    Password
  </label>
  <div
    class="control"
  >
    <input
      class="styled__Input-ms4xi4-0 fbXGEM"
      name="password"
      type="password"
      value=""
    />
  </div>
  
</div>
`;
